app/services/pii/cipher.rb

```diff
@@ -0,0 +1,39 @@
+module Pii
+  class Cipher
+    delegate :random_key, to: :cipher
+
+    def initialize
+      @cipher = OpenSSL::Cipher.new 'AES-256-CBC'
```

Comment:
I don't think you want to reuse the `OpenSSL::Cipher` object across multiple encrypts/decrypts since the `key`/`iv`/etc are all mutable state on it.